src/prompt/base.rs

```diff
@@ -97,22 +90,6 @@ pub trait Prompt: Send {
         &self,
         history_search: PromptHistorySearch,
     ) -> Cow<str>;
-    /// Get the default prompt color
-    fn get_prompt_color(&self) -> Color {
```

Comment:
I'm just wondering if there's a way to do this without creating a breaking change. Removing these 4 functions will break everyone who's using reedline and making custom prompts, right?

Reply:
I don't see a way to keep these functions.

These 4 functions have default implementations and are the current way to change the default colors of reedline.  Looking at [the reverse dependancies of reedline](https://crates.io/crates/reedline/reverse_dependencies) I picked a few to check if they have there own implementations ans so this would be a breaking change:

That don't: 

- [nu-cli](https://github.com/nushell/nushell/blob/main/crates/nu-cli/src/prompt.rs)
- [reedline-repl-rs](https://github.com/arturh85/reedline-repl-rs/blob/main/src/prompt.rs)
- [aichat](https://github.com/sigoden/aichat/blob/main/src/repl/prompt.rs)
- [Cecile](https://github.com/your-moon/Cecile/blob/master/src/repl/mod.rs)
- [dust_lang](https://git.jeffa.io/jeff/dust/src/branch/main/src/main.rs)

That do:

- [diatom-cli](https://github.com/diatom-lang/diatom/blob/master/diatom-cli/src/cli/prompt.rs)

Since changing the colors are already used by existing users, at the very least I would need to expose Theme record for diatom. If having a breaking change isn't a show stopper, I will add that.

Reply:
of course, I'm most concerned about nushell with this change.

Reply:
Glade this isn't a blocker. Added builder methods so diathom-cli (and other users) can still change the colors.